direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java

```diff
@@ -764,10 +774,23 @@ public void onPartitionsAssigned(Collection<TopicPartition> parts) {
                 entry -> {
                   final long offset = entry.getValue() == null ? 0L : entry.getValue().offset();
                   return new TopicOffset(
-                      entry.getKey().partition(), offset, clock.get().getWatermark());
+                      entry.getKey().partition(), offset, watermarkEstimator.get().getWatermark());
                 })
             .collect(Collectors.toList());
       }
+
+      private WatermarkEstimator createWatermarkEstimator() {
+        final WatermarkIdlePolicyFactory idlePolicyFactory =
+            accessor.getWatermarkConfiguration().getWatermarkIdlePolicyFactory();
+        final WatermarkEstimatorFactory estimatorFactory =
+            accessor.getWatermarkConfiguration().getWatermarkEstimatorFactory();
+        return estimatorFactory.create(cfg, idlePolicyFactory);
+      }
     };
   }
+
+  private static PartitionedWatermarkEstimator createWatermarkEstimatorForEmptyParts() {
```

Comment:
So far I cant remove processing time estimator for empty and unassigned partitions because lot of test fails. I make it more explicit, so I removed processing-time from PartitionedEstimator. 
If behaviour is not correct must be fixed by another issue.

Reply:
O2-Czech-Republic#155